tests/components/sentry/test_init.py

```diff
@@ -0,0 +1,330 @@
+        level=logging.WARNING, event_level=logging.WARNING
+    )
+
+    assert sentry_aiohttp_mock.call_count == 1
+    assert sentry_sqlalchemy_mock.call_count == 1
+    assert sentry_mock.init.call_count == 1
+
+    call_args = sentry_mock.init.call_args[1]
+    assert set(call_args) == {
+        "dsn",
+        "environment",
+        "integrations",
+        "release",
+        "before_send",
+    }
+    assert call_args["dsn"] == "http://public@example.com/1"
+    assert call_args["environment"] == "production"
+    assert call_args["integrations"] == [
+        sentry_logging_mock.return_value,
+        sentry_aiohttp_mock.return_value,
+        sentry_sqlalchemy_mock.return_value,
+    ]
+    assert call_args["release"] == current_version
+    assert call_args["before_send"]
+
+
+async def test_setup_entry_with_tracing(hass: HomeAssistant) -> None:
+    """Test integration setup from entry with tracing enabled."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_DSN: "http://public@example.com/1"},
+        options={CONF_TRACING: True, CONF_TRACING_SAMPLE_RATE: 0.5},
+    )
+    entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.sentry.AioHttpIntegration"), patch(
+        "homeassistant.components.sentry.SqlalchemyIntegration"
+    ), patch("homeassistant.components.sentry.LoggingIntegration"), patch(
+        "homeassistant.components.sentry.sentry_sdk"
+    ) as sentry_mock:
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    call_args = sentry_mock.init.call_args[1]
+    assert set(call_args) == {
+        "dsn",
+        "environment",
+        "integrations",
+        "release",
+        "before_send",
+        "traceparent_v2",
+        "traces_sample_rate",
+    }
+    assert call_args["traces_sample_rate"] == 0.5
+    assert call_args["traceparent_v2"]
+
+
+@pytest.mark.parametrize(
+    "version,channel",
+    [
+        ("0.115.0.dev20200815", "nightly"),
+        ("0.115.0", "stable"),
+        ("0.115.0b4", "beta"),
+        ("0.115.0dev0", "dev"),
+    ],
+)
+async def test_get_channel(version, channel) -> None:
+    """Test if channel detection works from Home Assistant version number."""
+    assert get_channel(version) == channel
+
+
+async def test_process_before_send(hass: HomeAssistant):
+    """Test regular use of the Sentry process before sending function."""
+    hass.config.components.add("puppies")
+    hass.config.components.add("a_integration")
+
+    # These should not show up in the result.
+    hass.config.components.add("puppies.light")
+    hass.config.components.add("auth")
+
+    result = process_before_send(
+        hass,
+        options={},
+        channel="test",
+        huuid="12345",
+        system_info={"installation_type": "pytest"},
+        custom_components=["ironing_robot", "fridge_opener"],
+        event={},
+        hint={},
+    )
+
+    assert result
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)